sql_file3A.sql

```diff
@@ -0,0 +1,163 @@
+(1, 1, 1, '2024-09-15'),
+(2, 2, 2, '2024-09-20'),
+(3, 3, 3, '2024-09-25'),
+(4, 4, 4, '2024-09-30');
+
+-- Questions
+
+-- 1. List all patients and their appointments, including patients who have never had an appointment.
+
+SELECT P.PatientID , P.PatientName, A.AppointmentId
+FROM Patients P
+LEFT JOIN  Appointments A ON P.PatientID = A.PatientID;
+
+-- 2. Display all doctors and their scheduled appointments, including doctors without any appointments.
+
+SELECT D.DoctorID , D.DoctorName, A.AppointmentId,A.AppointmentDate
+FROM Doctors D
+LEFT JOIN  Appointments A ON D.DoctorID = A.PatientID;
+
+-- 3. Show all medications and the patients they've been prescribed to, including medications that haven't been prescribed.
+
+SELECT M.MedicationID,M.MedicationName,M.DosageForm,P.PatientID,P.PatientName
+FROM Medications M
+LEFT JOIN Prescriptions R ON R.MedicationID = M.MedicationID
+LEFT JOIN Patients P ON P.PatientID = R.PatientID
+LEFT JOIN Appointments A ON A.PatientID = P.PatientID;
+
+-- 4. List all possible patient-doctor combinations, regardless of whether an appointment has occurred.
+
+SELECT *
+FROM Patients
+CROSS JOIN  Doctors;
+
+
+
+-- 5. Display all prescriptions with patient and medication information, including prescriptions where either the patient or medication information is missing.
+
+SELECT * 
+FROM Prescriptions R
+LEFT JOIN Patients P ON R.PatientID = P.PatientID
+LEFT JOIN Medications M ON R.MedicationID = M.MedicationID;
+
+-- 6. Show all patients who have never been prescribed any medication, along with their appointment history.
+
+SELECT P.PatientID,P.PatientName,P.DateOfBirth,P.Gender
+FROM Patients P
+LEFT JOIN Prescriptions R ON P.PatientID = R.MedicationID
+LEFT JOIN Medications M ON R.MedicationID = M.MedicationID
+
+WHERE M.MedicationID IS NULL;
+
+-- 7. List all doctors who have appointments in the next week, along with the patients they're scheduled to see.
+
+SELECT D.DoctorID,D.DoctorName,D.Specialty,P.PatientID,P.PatientName,A.AppointmentDate
+FROM Doctors D
+LEFT JOIN Appointments A ON D.DoctorID = A.DoctorID
+LEFT JOIN Patients P ON P.PatientID = A.PatientID
+WHERE DATEPART(WEEK,A.AppointmentDate) = DATEPART(WEEK,GETDATE()) +1;
+
+
+-- 8. Display all medications prescribed to patients over 60 years old, including medications not prescribed to this age group.
+
+SELECT M.MedicationID,M.MedicationName,M.DosageForm,P.PatientID,P.PatientName,P.DateOfBirth,P.Gender
+FROM Medications M
+LEFT JOIN Prescriptions R ON R.MedicationID = M.MedicationID
+LEFT JOIN Patients P ON P.PatientID = R.PatientID
+WHERE DATEDIFF(YEAR,P.DateOfBirth,GETDATE()) > 60;
+
+-- 9. Show all appointments from last year and any associated prescription information.
+
+SELECT A.AppointmentID,A.PatientID,A.DoctorID,A.AppointmentDate,R.PrescriptionID,R.PrescriptionDate,R.MedicationID
+FROM Appointments A
+LEFT JOIN Prescriptions R ON A.PatientID = R.MedicationID
```

Comment:
The join condition should be A.PatientID = R.PatientID, not R.MedicationID

Reply:
SELECT A.AppointmentID,A.PatientID,A.DoctorID,A.AppointmentDate,R.PrescriptionID,R.PrescriptionDate,R.MedicationID
FROM Appointments A
LEFT JOIN Prescriptions R ON A.PatientID = R.PatientID
WHERE YEAR(A.AppointmentDate) = YEAR(GETDATE())-1;